server/src/main/java/org/elasticsearch/search/rank/rrf/RRFRankDoc.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank.rrf;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * {@code RRFRankDoc} supports additional ranking information
+ * required for RRF.
+ */
+public class RRFRankDoc extends RankDoc {
+
+    public static final int NO_RANK = 0;
```

Comment:
Seems like `-1` would be better? Keeping things `0'd` indexed for consistency is nice.

Reply:
Done.